local/getDevice.go

```diff
@@ -5,59 +5,56 @@ import (
 	"fmt"
 	"sync"
 
-	"github.com/go-ocf/sdk/local/device"
-	"github.com/go-ocf/sdk/local/resource"
+	gocoap "github.com/go-ocf/go-coap"
+	"github.com/go-ocf/kit/net/coap"
+	"github.com/go-ocf/sdk/schema"
 )
 
-type deviceHandler struct {
-	deviceID string
-	cancel   context.CancelFunc
-
-	client *device.Client
-	lock   sync.Mutex
-	err    error
+// GetDevice performs a multicast and returns a device object if the device responds.
+func (c *Client) GetDevice(ctx context.Context, deviceID string) (*Device, error) {
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+	h := newDeviceHandler(deviceID, c.tlsConfig, c.conn, cancel)
+	err := DiscoverDevices(ctx, c.conn, h, coap.WithDeviceID(deviceID))
+	if err != nil {
+		return nil, fmt.Errorf("could not get the device %s: %v", deviceID, err)
+	}
+	d := h.Device()
+	if d == nil {
+		return nil, fmt.Errorf("no response from the device %s", deviceID)
+	}
+	return d, nil
 }
 
-func newDeviceHandler(deviceID string, cancel context.CancelFunc) *deviceHandler {
-	return &deviceHandler{deviceID: deviceID, cancel: cancel}
+func newDeviceHandler(deviceID string, tlsConfig *TLSConfig, multicastConn []*gocoap.MulticastClientConn, cancel context.CancelFunc) *deviceHandler {
+	return &deviceHandler{deviceID: deviceID, tlsConfig: tlsConfig, multicastConn: multicastConn, cancel: cancel}
 }
 
-func (h *deviceHandler) Handle(ctx context.Context, client *device.Client) {
-	h.lock.Lock()
-	defer h.lock.Unlock()
-	if client.DeviceID() == h.deviceID {
-		h.client = client
-		h.cancel()
-	}
-}
+type deviceHandler struct {
+	deviceID      string
+	tlsConfig     *TLSConfig
+	multicastConn []*gocoap.MulticastClientConn
```

Comment:
If `multicastConn` was eliminated from `Device` as mentioned later, it could be removed from here as well.